pkg/skaffold/apiversion/apiversion.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package apiversion
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type ReleaseTrack int
+
+const (
+	alpha ReleaseTrack = 0
+	beta  ReleaseTrack = 1
+	ga    ReleaseTrack = 2
+)
+
+type Version struct {
+	Major   int
+	Minor   int
+	Release ReleaseTrack
+}
+
+var re = regexp.MustCompile(`^skaffold/v(\d)(?:(alpha|beta)(\d))?$`)
+
+// ParseAPIVersion parses a string into a Version.
+func ParseVersion(v string) (*Version, error) {
```

Comment:
Here's an easier mapping to blang/semver

v1alpha3 -> 1.0.0-alpha.3
v1 -> 1.0.0

v2beta3 -> 2.0.0-beta.3
v2beta4 -> 2.0.0-beta.4

Reply:
Once we map it to blang/semver, we can remove the comparison logic and just reuse that library's

Reply:
My 2 cents: 
The two solution seem pretty comparable, but I like that the blang/semver version because it seems like that the K8s versioning subset _is_ a subset of semver, hence the abstraction is correct. 
The lib will take care of version ordering after the mapping - and for the other blang/semver functionality - like range parsing - will be available, only we'd have to do similar mapping but could be useful and then we don't have to write tests for our own ordering. 

One issue for example with Dan's approach already is that "v1" is not supported. "ga" is not following the K8S versioning either. By mapping `<alpha/beta>X` to Semver's pre-release numbers, this goes away automatically. 

One problem I see with our approach is that there is no minor version numbers - but we made that call when we went with the K8S versioning.